[full-ci] The password policy added. Bump reva (#7195)
owncloud/ocis: services/frontend/README.md

2.  Though usually not necessary, a database name and a database table can be configured for event stores if the event store supports this. Generally not applicable for stores of type `in-memory`. These settings are blank by default which means that the standard settings of the configured store apply.
3.  The frontend service can be scaled if not using `in-memory` stores and the stores are configured identically over all instances.
4.  When using `redis-sentinel`, the Redis master to use is configured via `FRONTEND_OCS_STAT_CACHE_STORE_NODES` in the form of `<sentinel-host>:<sentinel-port>/<redis-master>` like `10.10.0.200:26379/mymaster`.

## The password policy

> Password can contain any utf-8 characters. The password policy allows to adjust the minimum password length and minimum number of uppercase/lowercase, digits and

Reviewer: ```suggestion
Password can contain any utf-8 characters. The password policy allows to adjust the minimum password length and minimum number of uppercase/lowercase letters, digits and
```

special characters for password validation. The password policy can be adjusted by the environment variables below.

>   -   `PASSWORD_POLICIES_MIN_CHARACTERS` - Define the minimum password length.
>   -   `PASSWORD_POLICIES_MIN_LOWER_CASE_CHARACTERS` - Define the minimum number of uppercase letters.
>   -   `PASSWORD_POLICIES_MIN_UPPER_CASE_CHARACTERS` - Define the minimum number of lowercase letters.
>   -   `PASSWORD_POLICIES_MIN_DIGITS` - Define the minimum number of digits.
>   -   `PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS` - Define the minimum number of special characters.
>   -   `PASSWORD_POLICIES_SPECIAL_CHARACTERS` - Define the custom list of special characters that will be counted and compared to `PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS`. Special characters can be any utf-8 characters like !"#$%&'()*+,-./:;<=>?@[\]^_`{|}~ or other. If you want to define your own list of special characters you have to follow the environment variables escaping rules of
> the operating system.

Reviewer 2: ```suggestion
  -   `FRONTEND_PASSWORD_POLICIES_MIN_CHARACTERS` - Define the minimum password length.
  -   `FRONTEND_PASSWORD_POLICIES_MIN_LOWER_CASE_CHARACTERS` - Define the minimum number of uppercase letters.
  -   `FRONTEND_PASSWORD_POLICIES_MIN_UPPER_CASE_CHARACTERS` - Define the minimum number of lowercase letters.
  -   `FRONTEND_PASSWORD_POLICIES_MIN_DIGITS` - Define the minimum number of digits.
  -   `FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS` - Define the minimum number of characters from the special characters list below to be present.
  -   `FRONTEND_PASSWORD_POLICIES_SPECIAL_CHARACTERS` - Define the custom list of characters that will count as special characters for `FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS`.  
Special characters can be any UTF-8 characters like `!"#$%&'()*+,-./:;<=>?@[\]^_{|}~` or others. Note, though possible, avoid the \` (backtick). If you want to define your own list of special characters, you have to follow the environment variables escaping rules of the operating system.
```
Add backticks a special case, just to be sure we (and support) dont run into a mess.
The two blanks in the last list item main line are by purpose to render the next line as part of that list item
The backtick part renders properly in markdown.

> ## The password policy
> 
> Password can contain any utf-8 characters. The password policy allows to adjust the minimum password length and minimum number of uppercase/lowercase letters, digits and
> special characters for password validation. The password policy can be adjusted by the environment variables below.
>   -   `FRONTEND_PASSWORD_POLICIES_MIN_CHARACTERS` - Define the minimum password length.
>   -   `FRONTEND_PASSWORD_POLICIES_MIN_LOWER_CASE_CHARACTERS` - Define the minimum number of uppercase letters.
>   -   `FRONTEND_PASSWORD_POLICIES_MIN_UPPER_CASE_CHARACTERS` - Define the minimum number of lowercase letters.
>   -   `FRONTEND_PASSWORD_POLICIES_MIN_DIGITS` - Define the minimum number of digits.
>   -   `FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS` - Define the minimum number of characters from the special characters list below to be present.
>   -   `FRONTEND_PASSWORD_POLICIES_SPECIAL_CHARACTERS` - Define the custom list of characters that will count as special characters for `FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS`.  
> Special characters can be any UTF-8 characters like `!"#$%&'()*+,-./:;<=>?@[\]^_{|}~` or others. Note, though possible, avoid the \` (backtick). If you want to define your own list of special characters, you have to follow the environment variables escaping rules of the operating system.
> 
> The password policy currently impacts only public link password validation.

Reviewer 2: ```suggestion
## Password Policy

Note that the password policy currently impacts _only_ public link password validation.

With the password policy, mandatory criterias for the password can be defined via the environment variables listed below.

A password respectively special characters can contain any UTF-8 characters.

-   `FRONTEND_PASSWORD_POLICIES_MIN_CHARACTERS`  
Define the minimum password length.
-   `FRONTEND_PASSWORD_POLICIES_MIN_LOWER_CASE_CHARACTERS`  
Define the minimum number of uppercase letters.
-   `FRONTEND_PASSWORD_POLICIES_MIN_UPPER_CASE_CHARACTERS`  
Define the minimum number of lowercase letters.
-   `FRONTEND_PASSWORD_POLICIES_MIN_DIGITS`  
Define the minimum number of digits.
-   `FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS`  
Define the minimum number of characters from the special characters list below to be present.
-   `FRONTEND_PASSWORD_POLICIES_SPECIAL_CHARACTERS`  
Define the custom list of characters that will count as special characters.  
Special characters can be any UTF-8 characters like `!"#$%&'()*+,-./:;<=>?@[\]^_{|}~` or others. Note, though possible, avoid the \` (backtick). If you want to define your own list of special characters, you have to follow the environment variables escaping rules of the operating system.
```
After re-reading, I have further improved the text a bit and formatting to ease understanding and readbility.

2.  Though usually not necessary, a database name and a database table can be configured for event stores if the event store supports this. Generally not applicable for stores of type `in-memory`. These settings are blank by default which means that the standard settings of the configured store apply.
3.  The frontend service can be scaled if not using `in-memory` stores and the stores are configured identically over all instances.
4.  When using `redis-sentinel`, the Redis master to use is configured via `FRONTEND_OCS_STAT_CACHE_STORE_NODES` in the form of `<sentinel-host>:<sentinel-port>/<redis-master>` like `10.10.0.200:26379/mymaster`.

## Password Policy

Note that the password policy currently impacts _only_ public link password validation.

With the password policy, mandatory criteria for the password can be defined via the environment variables listed below.

> Generally, a password can contain any UTF-8 characters, however some characters are regarded as special since they are not used in ordinary texts. Which characters should be treated as special can be defined via a list of special characters.

Reviewer 2: ```suggestion
Generally, a password can contain any UTF-8 characters, however some characters are regarded as special since they are not used in ordinary texts. Which characters should be treated as special can be defined via a list of special characters.

Note that a password can have a maximum length of **72 bytes**. Depending on the alphbet used, a character is encoded as 1 to 4 bytes defining the maximum length of a password indirectly. While US-ASCII will only need one byte, Latin alphabets and also Greek or Cyrillic need two bytes. Three bytes are needed for characters like Chinese, Japanese and Korean etc.
```
@EParzefall pls do a language review

Reviewer: Done. Nice explanation.

3.  The frontend service can be scaled if not using `in-memory` stores and the stores are configured identically over all instances.
4.  When using `redis-sentinel`, the Redis master to use is configured via `FRONTEND_OCS_STAT_CACHE_STORE_NODES` in the form of `<sentinel-host>:<sentinel-port>/<redis-master>` like `10.10.0.200:26379/mymaster`.

## The password policy

Note that the password policy currently impacts _only_ public link password validation.

With the password policy, mandatory criteria for the password can be defined via the environment variables listed below.

Generally, a password can contain any UTF-8 characters, however some characters are regarded as special since they are not used in ordinary texts. Which characters should be treated as special can be defined via a list of special characters.

Note that a password can have a maximum length of **72 bytes**. Depending on the alphabet used, a character is encoded by 1 to 4 bytes, defining the maximum length of a password indirectly. While US-ASCII will only need one byte, Latin alphabets and also Greek or Cyrillic ones need two bytes. Three bytes are needed for characters in Chinese, Japanese and Korean etc.

> -   `FRONTEND_PASSWORD_POLICIES_MIN_CHARACTERS`
> Define the minimum password length.
> -   `FRONTEND_PASSWORD_POLICIES_MIN_LOWER_CASE_CHARACTERS`
> Define the minimum number of uppercase letters.
> -   `FRONTEND_PASSWORD_POLICIES_MIN_UPPER_CASE_CHARACTERS`
> Define the minimum number of lowercase letters.
> -   `FRONTEND_PASSWORD_POLICIES_MIN_DIGITS`
> Define the minimum number of digits.
> -   `FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS`
> Define the minimum number of characters from the special characters list below to be present.
> -   `FRONTEND_PASSWORD_POLICIES_SPECIAL_CHARACTERS`
> Define the custom list of characters that will count as special characters.
> Special characters can be any UTF-8 characters like `!"#$%&'()*+,-./:;<=>?@[\]^_{|}~` or others. Avoid using the \` (backtick) although it's possible to do so. If you want to define your own list of special characters, you have to follow the environment variables escaping rules of the operating system.

Reviewer 2: ```suggestion
-   `FRONTEND_PASSWORD_POLICIES_MIN_CHARACTERS`  
Define the minimum password length.
-   `FRONTEND_PASSWORD_POLICIES_MIN_LOWER_CASE_CHARACTERS`  
Define the minimum number of uppercase letters.
-   `FRONTEND_PASSWORD_POLICIES_MIN_UPPER_CASE_CHARACTERS`  
Define the minimum number of lowercase letters.
-   `FRONTEND_PASSWORD_POLICIES_MIN_DIGITS`  
Define the minimum number of digits.
-   `FRONTEND_PASSWORD_POLICIES_MIN_SPECIAL_CHARACTERS`  
Define the minimum number of characters from the special characters list below to be present.
-   `FRONTEND_PASSWORD_POLICIES_SPECIAL_CHARACTERS`  
Define the custom list of characters that will count as special characters.  
Special characters can be any UTF-8 characters like `!"#$%&'()*+,-./:;<=>?@[\]^_{|}~` or others. Avoid using the \` (backtick) although it's possible to do so. If you want to define your own list of special characters, you have to follow the environment variables escaping rules of the operating system.
```
Readding the two blanks on each main list item. This makes the following line to be rendered as new line but part of the list item.
